tests/unit/database_tests.py

```diff
@@ -1065,5 +1066,108 @@ def test_get_query_indexes(self):
+            {'bookmark': 1},                    # Should be a basestring
+            {'counts': 'blah'},                 # Should be a list
+            {'drilldown': 'blah'},              # Should be a list
+            {'group_field': ['blah']},          # Should be an basestring
+            {'group_limit': 'int'},             # Should be an int
+            {'group_sort': 'blah'},             # Should be a list
+            {'include_docs': 'blah'},           # Should be an boolean
+            {'limit': 'blah'},                  # Should be an int
+            {'ranges': 1},                      # Should be a dict
+            {'sort': 10},                       # Should be a basestring or list
+            {'stale': ['blah']},                # Should be a basestring
+            {'highlight_fields': 'blah'},       # Should be a list
+            {'highlight_pre_tag': ['blah']},    # Should be a basestring
+            {'highlight_post_tag': 1},          # Should be a basestring
+            {'highlight_number': ['int']},      # Should be an int
+            {'highlight_size': 'blah'},         # Should be an int
+            {'include_fields': 'list'},         # Should be a list
+        ]
+
+        for argument in test_data:
+            try:
+                self.db.get_search_result('searchddoc001', 'searchindex001',
+                                          'julia*', **argument)
+                self.fail('Above statement should raise an Exception')
+            except CloudantArgumentError as err:
+                self.assertTrue(str(err).startswith(
+                    'Argument {0} is not an instance of expected type:'.format(
+                        list(argument.keys())[0]
+                    )
+                ))
+
+    def test_get_search_result_with_invalid_query_type(self):
+        """
+        Test get_search_result by passing an invalid query type
+        """
+        try:
+            self.db.get_search_result('searchddoc001', 'searchindex001', ['blah'])
+            self.fail('Above statement should raise an Exception')
+        except CloudantArgumentError as err:
+            self.assertTrue(str(err).startswith(
+                'Argument query is not an instance of expected type:'
+            ))
+
+    def test_get_search_result_executes_search_query(self):
+        """
+        Test get_search_result executes a search query
+        """
+        self.populate_db_with_documents(100)
+        resp = self.db.get_search_result(
+            'searchddoc001',
+            'searchindex001',
+            'julia*',
+            limit=5,
+            include_docs=True
+        )
+        self.assertEqual(len(resp['rows']), 5)
+        self.assertEqual(resp['total_rows'], 100)
+        for row in resp['rows']:
+            self.assertIsNotNone(row['fields'])
+            self.assertTrue(row['id'].startswith('julia0'))
+            self.assertIsNotNone(row['order'])
+            self.assertIsNotNone(row['doc'])
```

Comment:
I don't really like stepping through the elements of the `resp` and performing asserts inside of the loop.  Would it be possible to compose an object that looks like what you expect the `resp` to look like based on the query call and then to use that object to do a direct comparison to the `resp`?  This is was what I've tried to do as much as possible in this library and what most of the other client libraries also attempt to do.  Same comment for the group_option test below.